diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index 087d7b0d42c65d854f35d36a4a822f42d18e9b3f..0000000000000000000000000000000000000000
diff --git a/dist/browser/es/src/index.d.ts b/dist/browser/es/src/index.d.ts
index 5e333fcf633689d6bc47058e13d7eb7be3a79905..7615711cce0426859a48e7b44ad988d41c8af574 100644
--- a/dist/browser/es/src/index.d.ts
+++ b/dist/browser/es/src/index.d.ts
@@ -10,7 +10,6 @@ declare global {
     interface Window {
         ReactNativeWebView?: WebView;
         sdkProvider: SDKProvider;
-        ethereum?: SDKProvider;
         mmsdk?: MetaMaskSDK;
         extension?: MetaMaskInpageProvider;
         extensions?: any[];
