diff --git a/CHANGELOG.md b/CHANGELOG.md
deleted file mode 100644
index a13db885bdd2be3b7fc0f92a37cd35137e470832..0000000000000000000000000000000000000000
diff --git a/dist/browser/es/src/index.d.ts b/dist/browser/es/src/index.d.ts
index 3c9ae3784ea4d11a09f28ff2e3c0433bf8a50d10..2826a97f0881e03f72380cb3f4c65fa53c0d7880 100644
--- a/dist/browser/es/src/index.d.ts
+++ b/dist/browser/es/src/index.d.ts
@@ -11,7 +11,6 @@ declare global {
     interface Window {
         ReactNativeWebView?: WebView;
         sdkProvider: SDKProvider;
-        ethereum?: SDKProvider;
         mmsdk?: MetaMaskSDK;
         extension?: MetaMaskInpageProvider;
         extensions?: any[];
